File: test_integration.py
#!/usr/bin/env python3
"""
Test script to verify the AI Calorie App integration
"""

import requests
import base64
import json
import time
from PIL import Image
import io

def test_api_health():
    """Test if the API is running and healthy"""
    try:
        response = requests.get("http://localhost:8000/health", timeout=5)
        if response.status_code == 200:
            data = response.json()
            print("✅ API Health Check: PASSED")
            print(f"   Status: {data.get('status')}")
            print(f"   Models Available: {data.get('models_available')}")
            return True
        else:
            print(f"❌ API Health Check: FAILED (Status: {response.status_code})")
            return False
    except requests.exceptions.RequestException as e:
        print(f"❌ API Health Check: FAILED (Connection Error: {e})")
        return False

def create_test_image():
    """Create a simple test image"""
    # Create a simple colored image for testing
    img = Image.new('RGB', (224, 224), color='red')
    
    # Convert to base64
    buffer = io.BytesIO()
    img.save(buffer, format='JPEG')
    img_data = buffer.getvalue()
    base64_img = base64.b64encode(img_data).decode('utf-8')
    
    return base64_img

def test_food_analysis():
    """Test the food analysis endpoint"""
    try:
        # Create test image
        test_image = create_test_image()
        
        # Test data
        test_data = {
            "image": test_image,
            "context": "This is a test image",
            "format": "image/jpeg"
        }
        
        print("🧪 Testing food analysis...")
        response = requests.post(
            "http://localhost:8000/api/analyze",
            json=test_data,
            timeout=30
        )
        
        if response.status_code == 200:
            data = response.json()
            print("✅ Food Analysis: PASSED")
            print(f"   Success: {data.get('success')}")
            print(f"   Food Items Found: {len(data.get('food_items', []))}")
            print(f"   Total Calories: {data.get('nutritional_data', {}).get('total_calories', 0)}")
            return True
        else:
            print(f"❌ Food Analysis: FAILED (Status: {response.status_code})")
            print(f"   Response: {response.text}")
            return False
            
    except requests.exceptions.RequestException as e:
        print(f"❌ Food Analysis: FAILED (Connection Error: {e})")
        return False

def test_nextjs_api():
    """Test the Next.js API route"""
    try:
        # This would test the Next.js API route at localhost:3000
        # For now, we'll just check if the port is accessible
        response = requests.get("http://localhost:3000", timeout=5)
        if response.status_code == 200:
            print("✅ Next.js Frontend: ACCESSIBLE")
            return True
        else:
            print(f"⚠️  Next.js Frontend: Status {response.status_code}")
            return False
    except requests.exceptions.RequestException as e:
        print(f"⚠️  Next.js Frontend: Not accessible ({e})")
        print("   This is normal if you haven't started the frontend yet")
        return False

def main():
    """Run all integration tests"""
    print("🧪 AI Calorie App - Integration Test")
    print("====================================")
    
    tests_passed = 0
    total_tests = 3
    
    # Test 1: API Health
    if test_api_health():
        tests_passed += 1
    
    print()
    
    # Test 2: Food Analysis
    if test_food_analysis():
        tests_passed += 1
    
    print()
    
    # Test 3: Frontend
    if test_nextjs_api():
        tests_passed += 1
    
    print()
    print("====================================")
    print(f"Tests Passed: {tests_passed}/{total_tests}")
    
    if tests_passed == total_tests:
        print("🎉 All tests passed! Your integration is working perfectly.")
    elif tests_passed >= 1:
        print("⚠️  Some tests passed. Check the failures above.")
    else:
        print("❌ All tests failed. Please check your setup.")
        print("\n💡 Troubleshooting:")
        print("1. Make sure the Python API is running: python start_api.py")
        print("2. Check your .env file has a valid GROQ_API_KEY")
        print("3. Ensure all dependencies are installed: pip install -r requirements.txt")
    
    return tests_passed == total_tests

if __name__ == "__main__":
    main()

File: test-integration.js
#!/usr/bin/env node

/**
 * Test Integration Script
 * Tests the connection between backend and Python models service
 */

const fs = require('fs');
const path = require('path');

async function testIntegration() {
  console.log('🧪 Testing Integration Between Backend and Python Models Service\n');

  // Test 1: Check Python Models Service
  console.log('1️⃣ Testing Python Models Service...');
  try {
    const response = await fetch('https://food-detection-models.onrender.com/health');
    const data = await response.json();
    
    if (data.status === 'healthy') {
      console.log('✅ Python Models Service: HEALTHY');
      console.log(`   - Models Available: ${Object.keys(data.model_availability || {}).length}`);
      console.log(`   - Port: ${data.port || 'N/A'}`);
    } else {
      console.log('❌ Python Models Service: UNHEALTHY');
    }
  } catch (error) {
    console.log('❌ Python Models Service: CONNECTION FAILED');
    console.log(`   Error: ${error.message}`);
  }

  // Test 2: Check Backend Service
  console.log('\n2️⃣ Testing Backend Service...');
  try {
    const response = await fetch('https://food-analyzer-backend.onrender.com/health');
    const data = await response.json();
    
    if (response.ok) {
      console.log('✅ Backend Service: HEALTHY');
      console.log(`   - Status: ${data.status || 'OK'}`);
    } else {
      console.log('❌ Backend Service: UNHEALTHY');
    }
  } catch (error) {
    console.log('❌ Backend Service: CONNECTION FAILED');
    console.log(`   Error: ${error.message}`);
  }

  // Test 3: Test Detection Endpoint
  console.log('\n3️⃣ Testing Detection Endpoint...');
  try {
    // Create a simple test image (1x1 pixel)
    const testImageBuffer = Buffer.from([
      0xFF, 0xD8, 0xFF, 0xE0, 0x00, 0x10, 0x4A, 0x46, 0x49, 0x46, 0x00, 0x01,
      0x01, 0x01, 0x00, 0x48, 0x00, 0x48, 0x00, 0x00, 0xFF, 0xDB, 0x00, 0x43,
      0x00, 0x08, 0x06, 0x06, 0x07, 0x06, 0x05, 0x08, 0x07, 0x07, 0x07, 0x09,
      0x09, 0x08, 0x0A, 0x0C, 0x14, 0x0D, 0x0C, 0x0B, 0x0B, 0x0C, 0x19, 0x12,
      0x13, 0x0F, 0x14, 0x1D, 0x1A, 0x1F, 0x1E, 0x1D, 0x1A, 0x1C, 0x1C, 0x20,
      0x24, 0x2E, 0x27, 0x20, 0x22, 0x2C, 0x23, 0x1C, 0x1C, 0x28, 0x37, 0x29,
      0x2C, 0x30, 0x31, 0x34, 0x34, 0x34, 0x1F, 0x27, 0x39, 0x3D, 0x38, 0x32,
      0x3C, 0x2E, 0x33, 0x34, 0x32, 0xFF, 0xC0, 0x00, 0x11, 0x08, 0x00, 0x01,
      0x00, 0x01, 0x01, 0x01, 0x11, 0x00, 0x02, 0x11, 0x01, 0x03, 0x11, 0x01,
      0xFF, 0xC4, 0x00, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0xFF, 0xC4,
      0x00, 0x14, 0x10, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xDA, 0x00, 0x0C,
      0x03, 0x01, 0x00, 0x02, 0x11, 0x03, 0x11, 0x00, 0x3F, 0x00, 0x8A, 0x00,
      0xFF, 0xD9
    ]);

    const testData = {
      model_type: 'yolo',
      image_data: testImageBuffer.toString('base64'),
      width: 1,
      height: 1
    };

    const response = await fetch('https://food-detection-models.onrender.com/detect', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(testData)
    });

    if (response.ok) {
      const result = await response.json();
      console.log('✅ Detection Endpoint: WORKING');
      console.log(`   - Success: ${result.success}`);
      console.log(`   - Processing Time: ${result.processing_time || 'N/A'}ms`);
    } else {
      console.log('❌ Detection Endpoint: FAILED');
      console.log(`   Status: ${response.status}`);
    }
  } catch (error) {
    console.log('❌ Detection Endpoint: CONNECTION FAILED');
    console.log(`   Error: ${error.message}`);
  }

  console.log('\n🎉 Integration Test Complete!');
  console.log('\n📋 Next Steps:');
  console.log('1. Update your Render backend environment variables');
  console.log('2. Add PYTHON_MODELS_URL=https://food-detection-models.onrender.com');
  console.log('3. Redeploy your backend service');
  console.log('4. Test the full integration from your frontend');
}

// Run the test
testIntegration().catch(console.error);

